### DISCUSSION TITLE
Should we put the KB on Wordpress?

### DISCUSSION THREAD

**Discussion by gilgongo** [jamulussoftware/jamulus#1132] [Documentation]
The stuff about the Knowledge Base in https://github.com/jamulussoftware/jamuluswebsite/discussions/316 made me think about the fact that I was looking at SourceForge PHP/MySQL the other day for hosting discussions - but might it be better used for the KB?

The knowledge base is not a critical part of the project, so if it went down or needed to move to another platform one day it wouldn't be much of a big deal. That was my main concern with using SF to host our discussions. Also, Wordpress is widely-used, understood, exportable and flexible, and crucially - we could set the permissions to "contributor" by default AND allow social logins for GitHub, Facebook etc. thereby achieving a community wiki by other means. 

![bitmoji](https://sdk.bitmoji.com/render/panel/214e2e2c-3a98-46f6-9b61-50c57a0d8fca-e453f381-f13d-466a-a20e-c03c786b333b-v1.png?transparent=1&palette=1&width=246)

**Comment by gilgongo** [comment]
Out of curiosity, I asked SF to add a vhost called knowledge.jamulus.io which now resolves to https://sourceforge.net/projects/llcon/ but I can't work out how to make that resolve to the web space. If I can, I'll see if I can install WP as a test.

**Comment by ann0see** [comment]
Interesting suggestion.

I think (and you probably also think so) that we should try to get as much as possible on GitHub. 

Yes, adding content is a bit more difficult since we have all the fit stuff around it. WordPress would be easier but is there maybe a better way to add community content? E.g. via Issues which get converted to .md files and uploaded to jamulus.io? That would make adding/editing content a lot easier.

**Reply by gilgongo** [reply]
If we can get a contribution mechanism with as low a barrier to entry as possible then that would be good. Whether that's on GitHub or not isn't so much of an issue as long as it's on our domain name. The old wiki was "sign up, click edit/add page". We could I think do the same with WordPress (and with no markdown needed), although I must admit I've not seen that actually done. I should be able to play about with a proof of concept soon though (I'd forgotten we were using an .htaccess redirect so that's why the new host wasn't working). I give it 50:50 that SF will work though :-)

**Reply by ann0see** [reply]
I wouldn’t give too much effort on WordPress (that would be another place we have to maintain, update,...) I‘d focus on GitHub (Actions)

**Reply by gilgongo** [reply]
OK well, see what you can do on the submission process for the KB using GitHub and we can take a view. 

Oh BTW don't know if you've [seen this](https://github.com/jamulussoftware/jamuluswebsite/discussions/316) but I had some related questions.

EDIT: Ok you gave it a thumbs up so I take it you agree :-) I'll re-do those templates then (this is also part of what I'm doing on the "contributing" re-write)

**Reply by ann0see** [reply]
Ok. Great! I'll keep this and the mentioned issue in mind.

**Reply by gilgongo** [reply]
Just tried installing WordPress on SF.  Not really very practical: slow, no ability to download/install plugin or theme upgrades (would need to be done manually over rsync), and (bit of a showstopper) no email so no password resets or any status notifications possible. Oh well. That's SourceForge out of the picture then :-( I'd still like to see what WP does though so will maybe set up a test on my own server.

**Reply by gilgongo** [reply]
@ann0see OK I've set it up here as a proof of concept. https://knowledge.jamulus.io

If you register (we can have social logins for FB, Google and Twitter, but would have to pay for Gitub and others unfortunately) you will be an Author, which will let you edit and delete your own stuff.

**Reply by ann0see** [reply]
Just wanted to create an account there, but unfortunately I mis-typed my e-mail address. Can you delete the account?

**Reply by gilgongo** [reply]
OK done that.

**Reply by ann0see** [reply]
Great. Thanks. Nevertheless I think we should have a look at an option to hack GH issues to generate the posts. 

The GitHub API doesn't seem to be too difficult: https://api.github.com/repos/jamulussoftware/jamulus/issues

gives back json.

**Reply by gilgongo** [reply]
OK sure. Let's see what that looks like.

**Reply by ann0see** [reply]
As we already talk of project management: Should we open an issue on this?

**Reply by gilgongo** [reply]
An issue on what, sorry?

**Reply by ann0see** [reply]
An issue on how to make it possible to use GitHub issues as frontend for the knowledge base. This would allow us to have an alternative to WP without needing more maintenance.

**Reply by gilgongo** [reply]
Oh OK, sure. The maintenance is minimal BTW. I've got about 30 WP installations on that server and they're all backed up nightly (10 day incrementals) to S3 and upgraded (base, plugins and themes) on a weekly cron job. And like I said, the KB isn't mission critical so if for whatever reason it went down for a bit or needed migrating it wouldn't be a big deal.

**Reply by ann0see** [reply]
Yes. But it's another place where people need to create accounts, would need another (jamulus like) template,...

**Reply by gilgongo** [reply]
Sure (although GH doesn't do social login). But let's see what using Issues is like and we can compare it with WP. At least  we know what the WP route would involve now.

**Reply by ann0see** [reply]
The GitHub issues thing would require some coding/internet research but should be possible with GH Actions.

**Reply by ann0see** [reply]
Basically this action the other way round: https://github.com/peter-evans/create-issue-from-file

Step 1:
https://github.community/t/get-contents-body-of-issue-in-action/136057
Step 2: https://github.com/finnp/create-file-action

Should not be too difficult to code.
